BUILDING.md

```diff
@@ -15,7 +15,7 @@ Other should follow the user instructions. See the [How to use](https://github.c
 If you plan to build from source and run tests, install the following as well:
 ```sh
  $ [sudo] apt-get install libgflags-dev libgtest-dev
```

Comment:
The section "If you plan to build from source and run tests" doesn't make much sense.

You can certainly build from source without having clang-5.0   (and I think gflags and gtest are also not a requirement anymore).

cmake should be in a section "if you plan to build with cmake", not where it is now.